src/utils/with-react.js

```diff
@@ -73,4 +79,61 @@ const applyDefaults = ceInst => {
 
 export { applyDefaults };
 
-export default createElement => val(createElement);
+const distributeProperties = ({ children, ...properties }, componentClass) => {
```

Comment:
@markus-walther you are deconstructioning children and as return you are creating a new object with it. Why not deconstructing it directly here: 

IS: 
```
const reactStatelessComponent = properties => {
    const { attrs, props, children } = distributeProperties(
      properties,
      componentClass
    );

    return val(createElement)(
      tagName,
      { isReact: true, attrs, ...props },
      children
    );
  };
```

SUGGESTED: 
```
const reactStatelessComponent = ({children, ...properties}) => {
    const { attrs, props } = distributeProperties(
      properties,
      componentClass
    );

    return val(createElement)(
      tagName,
      { isReact: true, attrs, ...props },
      children
    );
  };
```

Reply:
Arguably the complexity is the same, no lines have been saved. But in the original solution the name distributeProperties is honoured more, since incoming children is a property that only via that function gets a special role.

Reply:
the advantage of my suggestion is that distributeProperties does not get polluted with props it doesn't need

Reply:
And please don't mark as resolved when its not resolved.

Reply:
If you refuse to add my improvement, lets ask the team